### PR TITLE
Fix text scaling for accessibility compliance

### PR DESCRIPTION
**Pull request by RafiKDev00**
### What's been done
 - Replace hardcoded font sizes with semantic fonts in iOS18/26 views  according to their equivalents from https://gist.github.com/zacwest/916d31da5d03405809c4 
 - Removed restrictive lineLimit constraints causing truncation in IOS26
 - Text now scales with Dynamic Type with standard accessibility text sizes (XS-XXXL) text on iOS 18/26 

### Known Issues
- iOS 18 still has layout overflow at extreme text sizes (needs scrolling implementation?), IOS26 truncates the messages
- Truncation begins at font size AX2 on an iPhone17 with IOS26, and screen overflow on iPhone17 with IOS18 at font size AX4
- Questions about optimal approach for handling very large text in the different IOS's (and should they be standardized between them)

###What's Next
 - Further work is required in order to scale the notification and fonts on Apples expanded accessibility settings, but this covers the standard sizes (XS through XXXL font), 
 - Standardize responses to extra large text
 - Scroll view? Will mess with Apples System logs today to get their dynamic
 - Bugs, edge cases, bolded, etc.
 - IOS17 if we have time...

### CODE REVIEW

File: Sources/UnionNotifications/NotificationDialogButton.swift
Comment (RafiKDev00) [inline Sources/UnionNotifications/NotificationDialogButton.swift:1]: Moved button Logic out of main code in order to shorten it and add readability. Also - hopefully can be abstracted now and used for IOS26.

File: Sources/UnionNotifications/NotificationDialogConstants..swift
Comment (RafiKDev00) [inline Sources/UnionNotifications/NotificationDialogConstants..swift:1]: These are the sizes we used to control the Dialog specs. Again abstracted to keep size of main IOS18NotifcationView file down, and also with the hope it can be used for ios26

File: Sources/UnionNotifications/NotificationDialogContent.swift
Comment (RafiKDev00) [inline Sources/UnionNotifications/NotificationDialogContent.swift:1]: These physical content of the message itself. Once more - abstracted to keep size of main IOS18NotifcationView file down, and also with the hope it can be used for ios26

File: Sources/UnionNotifications/iOS18NotificationView.swift
Comment (RafiKDev00) [inline Sources/UnionNotifications/iOS18NotificationView.swift:1]: the "main". I've cut it down significantly in order to abstract elements so they can (hopefully) be applied to IOS26. It should be readable now. Creates very convincing "apple" dialogs. Have to do some more testing to catch any possible weird looking edge cases - particularly in the iPad.

File: Sources/UnionNotifications/File.txt
Comment (RafiKDev00) [inline Sources/UnionNotifications/File.txt:1]: Pretty much same deal as above. Garbage file accidentally committed - was using it to store a block of code. I thought I had already deleted it locally. Won't be in next push.

File: Sources/UnionNotifications/File.swift
Comment (RafiKDev00) [inline Sources/UnionNotifications/File.swift:1]: Garbage file created for some purpose. Thought it had been deleted. Will not be in the next push.

File: Sources/UnionNotifications/File.swift [hunk 1]
```diff
@@ -0,0 +1,8 @@
+//
+//  File.swift
```
Comment (blsage) [inline Sources/UnionNotifications/File.swift]: I always try to update the comment to match the filename

File: Sources/UnionNotifications/File.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+VStack(spacing: 4) {
+    Text(LocalizedStringKey(localizedTitle))
+        .font(.body.weight(.semibold))
```
Comment (blsage) [inline Sources/UnionNotifications/File.txt]: Make sure that these semantic fonts are exactly the same as what the system dialog uses. One thing I did in the initial version of the component was to overlay a 0.5 opacity version of our custom component on top of the system dialog to dial it in.

File: Sources/UnionNotifications/File.txt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        .fixedSize(horizontal: false, vertical: true)
+
+    Text("Notifications may include alerts, sounds, and icon badges. These can be configured in Settings.", bundle: .module)
+        .font(.footnote.weight(.regular))
+        .foregroundColor(.primary)
+        .multilineTextAlignment(.center)
+        .fixedSize(horizontal: false, vertical: true)
+}
+.padding(.top, 20)
+.padding(.bottom, 8)  // Reduced bottom padding
```
Comment (blsage) [inline Sources/UnionNotifications/File.txt]: Generally speaking don't leave inline comments unless it's extremely unclear why a line would be necessary.

File: Sources/UnionNotifications/NotificationDialogButton.swift [hunk 4]
```diff
@@ -0,0 +1,37 @@
+//
+//  NotificationDialogButton.swift
+//  union-notifications
+//
+//  Created by RJ Kigner on 9/29/25.
+//
+
+import Foundation
+
```
Comment (blsage) [inline Sources/UnionNotifications/NotificationDialogButton.swift:7]: Imports should all be in a single stack, no returns. If you really want to worry about code style (or if the stack gets especially long) the convention is to order them alphabetically.

File: Sources/UnionNotifications/NotificationDialogButton.swift [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import SwiftUI
+import UnionButtons
+
+struct NotificationDialogButton: View {
+    let title: LocalizedStringKey
+    let isEnabled: Bool
```
Comment (blsage) [inline Sources/UnionNotifications/NotificationDialogButton.swift]: Whether or not a view is enabled in SwiftUI is handled with an environment property called `isEnabled`. You access it by declaring an environment variable:

```swift
@Environment(\.isEnabled) var isEnabled
```

And you set the property with the environment view modifier:

```swift
.environment(\.isEnabled, true)
```

Most environment properties in SwiftUI have preferred view modifiers that should be used instead of setting them directly. In the case of enabled/disabled:

```swift
.disabled(condition)
```

So because of this, you should never pass in a parameter regarding whether or not the view is enabled directly. Using the environment has many advantages. One obvious one is that you can disable an entire group by applying the parameter at the top level, and all of the children will receive the property value.

File: Sources/UnionNotifications/NotificationDialogButton.swift [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    let height: CGFloat
```
Comment (blsage) [inline Sources/UnionNotifications/NotificationDialogButton.swift]: Like enabled, framing should never be passed directly into a view. Instead, you should frame the outside of a component with the `.frame(width:height:)` view modifier.

File: Sources/UnionNotifications/iOS18NotificationView.swift [hunk 7]
```diff
@@ -1,79 +1,124 @@
...
-                        label
-                            .frame(maxWidth: .infinity, minHeight: 44)
-                            .background(
-                                Rectangle()
-                                    .fill(Color.primary.opacity(isPressed ? 0.1 : 0))
-                            )
-                    })
-                }
-                .frame(height: 44)
+                // Content area
```
Comment (blsage) [inline Sources/UnionNotifications/iOS18NotificationView.swift]: I tend to disfavor these comments and prefer self-commenting code

File: Sources/UnionNotifications/iOS18NotificationView.swift [hunk 8]
```diff
@@ -1,79 +1,124 @@
...
+                NotificationDialogContent(
+                    appName: appName,
+                    shouldEnableScroll: shouldEnableScroll,
+                    maxContentHeight: NotificationDialogConstants.maxScrollHeight
+                )
+                
+                // Button area
+                buttonSection
+            }
+            .background {
```
Comment (blsage) [inline Sources/UnionNotifications/iOS18NotificationView.swift]: `.background(.regularMagerial)`

Check out the API declaration for the background modifier. It's very dynamic. It can take any `ShapeStyle` which is a protocol that tons of different types of views and shaders conform to.

File: Sources/UnionNotifications/iOS18NotificationView.swift [hunk 9]
```diff
@@ -1,79 +1,124 @@
...
+                Rectangle()
+                    .fill(.regularMaterial)
             }
+            .frame(width: dialogWidth(
+                screenWidth: geometry.size.width,
+                screenHeight: geometry.size.height
+            ))
+            .clipShape(RoundedRectangle(cornerRadius: 16, style: .continuous))
```
Comment (blsage) [inline Sources/UnionNotifications/iOS18NotificationView.swift]: This works but is redundant. You can accomplish the same thing by saying

```swift
.clipShape(.rect(cornerRadius: 16)
```

File: Sources/UnionNotifications/iOS18NotificationView.swift [hunk 10]
```diff
@@ -1,79 +1,124 @@
...
+                )
+                
+                // Button area
+                buttonSection
+            }
+            .background {
+                Rectangle()
+                    .fill(.regularMaterial)
             }
+            .frame(width: dialogWidth(
```
Comment (blsage) [inline Sources/UnionNotifications/iOS18NotificationView.swift]: This is kinda awkward. I would instead do something like

```swift
.frame(width: dialogWidth(proxy: geometry)
```

File: Sources/UnionNotifications/iOS18NotificationView.swift [hunk 11]
```diff
@@ -1,79 +1,124 @@
...
+            ))
+            .clipShape(RoundedRectangle(cornerRadius: 16, style: .continuous))
+            .frame(maxWidth: .infinity, maxHeight: .infinity)
         }
-        .background {
-            Rectangle()
-                .fill(.regularMaterial)
+    }
+    
+    // MARK: Button Section!
```
Comment (blsage) [inline Sources/UnionNotifications/iOS18NotificationView.swift]: Unnecessary mark comment. If a view gets big, it's usually a sign to me that I should refactor into multiple views.

**Review by blsage — CHANGES_REQUESTED** [review]

File: Sources/UnionNotifications/File.swift (hunk 1 above)
Comment (blsage) [reply Sources/UnionNotifications/File.swift]: Why is the filename just `File.swift`?

**Review by blsage — CHANGES_REQUESTED** [review]

File: Sources/UnionNotifications/File.swift (hunk 1 above)
Comment (RafiKDev00) [reply Sources/UnionNotifications/File.swift]: With you on the naming, accidentally created the file. I didn't know that deleting a file on Xcode only deletes its reference, and they stay on the disk and track to git. Still thought I deleted them from the git before committing..but I'll be extra careful in the future.

File: Sources/UnionNotifications/File.txt (hunk 2 above)
Comment (RafiKDev00) [reply Sources/UnionNotifications/File.txt]: Good call with the overlay, I'll try that. This file was meant to be deleted, didn't realize that deleting on Xcode only removes the reference, and it still can get pushed. Eitherway actual stuff related to text is found in NotificationContent below.

File: Sources/UnionNotifications/iOS18NotificationView.swift (hunk 8 above)
Comment (RafiKDev00) [reply Sources/UnionNotifications/iOS18NotificationView.swift]: I was wondering about that, it was used somewhere in the original IOS18NotificationView code, so kept the convention.

File: Sources/UnionNotifications/iOS18NotificationView.swift (hunk 9 above)
Comment (RafiKDev00) [reply Sources/UnionNotifications/iOS18NotificationView.swift]: I changed it. I'm keeping the .continuous styling, I can't really tell the difference at this scale, but supposedly it's more IOS like than the default, so safer for the long run.

File: Sources/UnionNotifications/iOS18NotificationView.swift (hunk 11 above)
Comment (RafiKDev00) [reply Sources/UnionNotifications/iOS18NotificationView.swift]: With my property declarations all the marks become redundant. I'm used to commenting. I'll get rid of them.

File: Sources/UnionNotifications/NotificationDialogButton.swift [hunk 12]
```diff
@@ -22,7 +22,7 @@ struct NotificationDialogButton: View {
         .contentShape(.rect)
         .buttonStyle(UnionButtonStyle(nil) { label, isPressed in
             label
-                .frame(maxWidth: .infinity)  // No height here either
```
Comment (RafiKDev00) [inline Sources/UnionNotifications/NotificationDialogButton.swift]: Old habits die hard with the comments.

File: Sources/UnionNotifications/iOS18NotificationView.swift [hunk 13]
```diff
@@ -106,19 +99,21 @@ struct iOS18NotificationView: View {
...
     }
     
     private var activeButton: some View {
         NotificationDialogButton(
             title: "Allow",
-            isEnabled: true,
-            height: buttonHeight,
             action: onAllow
-        )
+        ).frame(minHeight: buttonHeight)
```
Comment (RafiKDev00) [inline Sources/UnionNotifications/iOS18NotificationView.swift]: Moved framing out of the view per comment

File: Sources/UnionNotifications/NotificationDialogButton.swift [hunk 14]
```diff
@@ -5,29 +5,24 @@
 //  Created by RJ Kigner on 9/29/25.
 //
 
-import Foundation
```
Comment (RafiKDev00) [inline Sources/UnionNotifications/NotificationDialogButton.swift]: fixing formatting (and got rid of unnecessary import)

File: Sources/UnionNotifications/NotificationDialogButton.swift [hunk 15]
```diff
@@ -5,29 +5,24 @@
...
-
 import SwiftUI
 import UnionButtons
 
+
```
Comment (RafiKDev00) [inline Sources/UnionNotifications/NotificationDialogButton.swift:11]: No longer passing in .isEnabled or framing.

File: Sources/UnionNotifications/iOS18NotificationView.swift (hunk 9 above)
Comment (blsage) [reply Sources/UnionNotifications/iOS18NotificationView.swift]: SwiftIU rounded rectangle is continuous corner radius by default. You can check the header declaration if you want:

```swift
@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
extension Shape where Self == RoundedRectangle {

    /// A rectangular shape with rounded corners, aligned inside the frame of
    /// the view containing it.
    public static func rect(cornerSize: CGSize, style: RoundedCornerStyle = .continuous) -> Self

    /// A rectangular shape with rounded corners, aligned inside the frame of
    /// the view containing it.
    public static func rect(cornerRadius: CGFloat, style: RoundedCornerStyle = .continuous) -> Self
}
```

File: Sources/UnionNotifications/iOS18NotificationView.swift (hunk 13 above)
Comment (blsage) [reply Sources/UnionNotifications/iOS18NotificationView.swift]: Make sure each of your view modifiers is always on a new line

File: Sources/UnionNotifications/iOS18NotificationView.swift [hunk 16]
```diff
@@ -1,79 +1,132 @@
...
 //  union-notifications
 //
 //  Created by Ben Sage on 9/23/25.
-//
+//  Last Update Rafi Kigner 9/30/25
 
 import SwiftUI
 import UnionButtons
 
+
```
Comment (blsage) [inline Sources/UnionNotifications/iOS18NotificationView.swift:10]: Stray newline

File: Sources/UnionNotifications/iOS18NotificationView.swift [hunk 17]
```diff
@@ -1,79 +1,132 @@
...
 struct iOS18NotificationView: View {
+
     let appName: String
     let onAllow: () -> Void
+    let onDontAllow: (() -> Void)? 
+    
+    @Environment(\.dynamicTypeSize) private var dynamicTypeSize
+    @Environment(\.horizontalSizeClass) private var horizontalSizeClass
+    
+    init(
```
Comment (blsage) [inline Sources/UnionNotifications/iOS18NotificationView.swift]: Problematic indentation on this init block

File: Sources/UnionNotifications/iOS18NotificationView.swift [hunk 18]
```diff
@@ -1,79 +1,132 @@
...
+           appName: String,
+           onAllow: @escaping () -> Void,
+           onDontAllow: (() -> Void)? = nil
```
Comment (blsage) [inline Sources/UnionNotifications/iOS18NotificationView.swift]: Don't allow will need to be escaping as well

File: Sources/UnionNotifications/iOS18NotificationView.swift [hunk 19]
```diff
@@ -1,79 +1,132 @@
...
+        HStack(spacing: 0) {
+            disabledButton
+            Divider()
+            activeButton
+        }
+        .frame(height: buttonHeight)
+    }
+    
+    private var disabledButton: some View {
+        Button(action: {
```
Comment (blsage) [inline Sources/UnionNotifications/iOS18NotificationView.swift]: Always use trailing closures

File: Sources/UnionNotifications/iOS18NotificationView.swift [hunk 20]
```diff
@@ -1,79 +1,132 @@
...
+                onDontAllow?()  
+            }) {
+            Text("Don't Allow", bundle: .module)
```
Comment (blsage) [inline Sources/UnionNotifications/iOS18NotificationView.swift]: Make sure that your strings are using angled quotes. The system dialog uses these, so it's implicit in the spec.

File: Sources/UnionNotifications/iOS18NotificationView.swift [hunk 21]
```diff
@@ -82,3 +135,10 @@ struct iOS18NotificationView: View {
         print("Allow tapped")
     }
 }
+#Preview {
+    iOS18NotificationView(
+        appName: "TestApp",
+        onAllow: { print("Allow") },
```
Comment (blsage) [inline Sources/UnionNotifications/iOS18NotificationView.swift]: Use multiple trailing closures

File: Sources/UnionNotifications/IOS18NotificationDialogContent.swift [hunk 22]
```diff
@@ -0,0 +1,52 @@
...
+//  IOS18NotificationDialogContent.swift
+//  union-notifications
+//
+//  Created by RJ Kigner on 9/29/25.
+//
+
+import Foundation
+import SwiftUI
+
+struct  IOS18NotificationDialogContent: View {
```
Comment (blsage) [inline Sources/UnionNotifications/IOS18NotificationDialogContent.swift]: Extra space

File: Sources/UnionNotifications/IOS18NotificationDialogContent.swift [hunk 23]
```diff
@@ -0,0 +1,52 @@
...
+                .font(.footnote.weight(.regular))
+                .foregroundColor(.primary)
+                .multilineTextAlignment(.center)
+                .fixedSize(horizontal: false, vertical: true)
+        }
+        .padding(.top, 20)
+        .padding(.bottom, 8)
+        .padding(.horizontal, 16)
+    }
+}
```
Comment (blsage) [inline Sources/UnionNotifications/IOS18NotificationDialogContent.swift]: Every view should have a preview

File: Sources/UnionNotifications/IOS18NotificationDialogContent.swift [hunk 24]
```diff
@@ -0,0 +1,52 @@
...
+    let appName: String
+    let shouldEnableScroll: Bool
+    let maxContentHeight: CGFloat?
+    
+    private var localizedTitle: String {
+        String(format: String(localized: "\"%@\" Would Like to Send You Notifications", bundle: .module), appName) 
+    }
+    
+    var body: some View {
+        Group {
```
Comment (blsage) [inline Sources/UnionNotifications/IOS18NotificationDialogContent.swift]: `body` uses `@ViewBuilder` by default, so you don't need to wrap this control flow in a `Group`.

File: Sources/UnionNotifications/IOS18NotificationDialogContent.swift [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+    private var contentText: some View {
+        VStack(spacing: 4) {
+            Text(LocalizedStringKey(localizedTitle))
+                .font(.body.weight(.semibold))
+                .foregroundColor(.primary)
+                .multilineTextAlignment(.center)
+                .lineSpacing(2)
+                .fixedSize(horizontal: false, vertical: true)
+
+            Text("Notifications may include alerts, sounds, and icon badges. These can be configured in Settings.", bundle: .module)
```
Comment (blsage) [inline Sources/UnionNotifications/IOS18NotificationDialogContent.swift]: Set a maximum column length for your editor (something like 80 or 100 or 120) and make sure to adhere to it. You can look up multiline strings if needed.

File: Sources/UnionNotifications/iOS26NotificationView.swift [hunk 26]
```diff
@@ -1,71 +1,133 @@
...
+    }
+    
+    private var horizontalButtons: some View {
+        HStack(spacing: 8) {
+            disabledButton
+            activeButton
+        }
+    }
+    
+    @Environment(\.colorScheme) private var colorScheme
```
Comment (blsage) [inline Sources/UnionNotifications/iOS26NotificationView.swift]: Your environment properties should always be declared at the top of the view.

File: Sources/UnionNotifications/iOS26NotificationView.swift [hunk 27]
```diff
@@ -1,71 +1,133 @@
...
+    
+    private var disabledButton: some View {
+        Button(action: { }) {
```
Comment (blsage) [inline Sources/UnionNotifications/iOS26NotificationView.swift]: Use multiple trailing closures

File: Sources/UnionNotifications/iOS26NotificationView.swift [hunk 28]
```diff
@@ -1,71 +1,133 @@
...
+                .font(.body.weight(.medium))
+                .foregroundStyle(.secondary)
+                .frame(maxWidth: .infinity, minHeight: buttonHeight)
+        }
+        .background(.gray.opacity(0.15), in: .rect(cornerRadius: 30, style: .continuous))
+        .glassEffect(.clear, in: .rect(cornerRadius: 30))
+        .disabled(true)
+    }
+    
+
```
Comment (blsage) [inline Sources/UnionNotifications/iOS26NotificationView.swift:220]: Stray newline

File: Sources/UnionNotifications/iOS26NotificationView.swift [hunk 29]
```diff
@@ -1,71 +1,133 @@
...
+    private var activeButton: some View {
+        Button(action: onAllow) {
+            Text("Allow", bundle: .module)
+                .font(.body.weight(.semibold))
+                .foregroundStyle(colorScheme == .dark ? .white : .black)
+                .frame(maxWidth: .infinity, minHeight: buttonHeight)
         }
-        .frame(width: 320)
-        .glassEffect(.regular, in: .rect(cornerRadius: 32))
+        .background(.gray.opacity(0.15), in: .rect(cornerRadius: 30, style: .continuous))
```
Comment (blsage) [inline Sources/UnionNotifications/iOS26NotificationView.swift]: How sure are you about this gray opacity effect? And how did you identify this? I would be pretty surprised if Apple were using a gray overlay combined with a clear effect rather than a regular glass effect for a system dialog. But if you tested extensively then no problem.

File: Sources/UnionNotifications/NotificationDialogButton.swift [hunk 30]
```diff
@@ -0,0 +1,32 @@
...
+    let action: () -> Void
+    
+    var body: some View {
+        Button(action: action) {
+            Text(title, bundle: .module)
+                .font(.body.weight(.semibold))
+                .foregroundStyle(.blue)
+        }
+        .contentShape(.rect)
+        .buttonStyle(UnionButtonStyle(nil) { label, isPressed in
```
Comment (blsage) [inline Sources/UnionNotifications/NotificationDialogButton.swift:23]: Union button style should only be used on iOS 18. It's unnecessary on iOS 26.

**Review by blsage — CHANGES_REQUESTED** [review]
